### DISCUSSION TITLE
Httpx is not encoding "%" symbol while sending requests

### DISCUSSION THREAD

**Discussion by heysarthak** [encode/httpx#3094] [Potential Issue]
Hi team, 

I am ussing httpx plugin, while sending httpx query "ngi%abc.com" .
ideally "%" symbol should be encoded as "%25" but it is still sending as the same "%".

can we please have a fix for this.

**Comment by lovelydinosaur** [comment]
Could you be more specific and demonstrate the actual code?

**Reply by heysarthak** [reply]
`self.conn=httpx.Client(http2=True,http1=False,proxies=self.proxies,verify=False,timeout=10.0)`

While sending the above command in httpx logs i can find 

httpx:HTTP Request: DELETE http://<>.ats-ravi/nudr-dr-prov/v2/policy-data/nai-eng%nai2@example.net "HTTP/2 400 Bad Request"

but the request should be http://<>.ats-ravi/nudr-dr-prov/v2/policy-data/nai-eng%25nai2@example.net as "%" symbol should be encoded as "%25".

**Reply by heysarthak** [reply]
Hi @tomchristie ,
do you need more details? 
was i able to answer your question?

thanks

**Reply by heysarthak** [reply]
Hi @tomchristie,

Should i move it to a bug?

**Reply by lovelydinosaur** [reply]
Hmmm, so the URL `https://www.example.com/test%this` doesn't need escaping.

(Eg. if you try pasting it into a browser bar it'll be sent as-is, unlike `https://www.example.com/test this`)

**Reply by heysarthak** [reply]
the url <>.ats-ravi/nudr-dr-prov/v2/policy-data/nai-eng%nai2@example.net SHOULD BE SENT AS
http://<>.ats-ravi/nudr-dr-prov/v2/policy-data/nai-eng%25nai2@example.net

if you write on google "searching for %"

the query that is sent is "https://www.google.com/search?q=searching+for+%25"
where % is being encoded as %25

But this is not happening with this release of package. v0.24.0+

it used to work with 
[Version 0.23.3]

**Reply by heysarthak** [reply]
@tomchristie do you have any solution to this?

**Comment by gyanu2507** [comment]
httpx treats `%` in a URL as already percent-encoded, so `ngi%abc.com` goes out with a literal `%`. That matches RFC 3986 (a `%` starts an escape). Requests has the same idea in most cases.

If you want the byte `%` on the wire as `%25`:

```python
from urllib.parse import quote

httpx.get("https://example.com/search", params={"q": "ngi%abc.com"})
# params are encoded, so this becomes q=ngi%25abc.com

# for a path segment:
path = quote("ngi%abc.com", safe="")
httpx.get(f"https://example.com/{path}")
```

Do not put a raw `%` in a path string and expect httpx to guess. Tom asked for a snippet of the actual call. If you paste that, we can say whether it is path, query, or a raw URL.
